dll_tests.js

```diff
@@ -0,0 +1,323 @@
+(function () {
+    function assert (condition, message) {
+    
+        if (condition) {
+            throw new Error(message);
+        }
+    }
+
+    function assertEqual (reference, value) {
+
+        if (reference !== value) {
+
+            throw new Error(reference + ' !== ' + value);
+        }
+    }
+    
+    function assertError(statement, errorMessage) {
+        
+        try {
+            eval(statement);
+        } catch (error) {
+            if(error.toString().search(errorMessage) > -1) {
+                return;
+            } 
+            else {
+                throw new Error('Error should be trown');
+            }
+        }
+    }
+
+    var unitTests = {
+
+        dLinkedListCreationTest: function () {
+
+            var dLinkedList = DLinkedList();
+
+            assert(dLinkedList.head && dLinkedList._size && dLinkedList.tail, 'dLinkedList was not created');
+        },
+
+        insertLeftTest: function () {
+
+            var dLinkedList = DLinkedList();
+
+            dLinkedList.insertLeft(5);
+            assertEqual(dLinkedList.head.data, 5);
+            assertEqual(dLinkedList.head, dLinkedList.tail);
+            assertEqual(dLinkedList._size, 1);
+
+            dLinkedList.insertLeft(3);
+            assertEqual(dLinkedList.head.data, 3);
+            assertEqual(dLinkedList.tail.data, 5);
+            assertEqual(dLinkedList.head.next.data, dLinkedList.tail.data);
+            assertEqual(dLinkedList.tail.previous.data, dLinkedList.head.data);
+            assertEqual(dLinkedList._size, 2);
+            
+             dLinkedList.insertLeft('string');
+            assertEqual(dLinkedList.head.data, 'string');
+            assertEqual(dLinkedList.tail.data, 5);
+            assertEqual(dLinkedList.head.next.data, 3);
+            assertEqual(dLinkedList.tail.previous.data, 3);
+            assertEqual(dLinkedList._size, 3);
+            
+            console.log('insertLeftTest OK');
```

Comment:
This could be done automatically from the for loop, also you should have something saying the unittest started.
`console.log` should not be used in the middle of a function like this if possible
you should not repeat yourself, here you are copy pasting the name of the function `insertLeftTest` into the log, it means if you want to change the name of the function you have to know that you also need to change the inside of the function.